src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -9234,14 +9410,14 @@ void Assembler::pusha() { // 32bit
 }
 
 void Assembler::set_byte_if_not_zero(Register dst) {
-  emit_int24(0x0F, (unsigned char)0x95, (0xE0 | dst->encoding()));
+  emit_int24(0x0F, (unsigned char)0x95, (0xC0 | dst->encoding()));
```

Comment:
Can you explain this change?
I see it is used only in c1_LIRAssembler_x86.cpp

Reply:
The operand encoding of SETNE r/m8 is just 0b11.000.000 AFAIU, and not 0b11.100.000 as in the previous code. I found this difference while comparing the bytes emitted by instructions encoded using "opcode(...)" with instructions emitted by MacroAssembler.

Reply:
I.e., I basically patched the MacroAssembler routine to emit an instruction that matched this: https://github.com/openjdk/jdk/blob/b84f6901275aedfeb59438898846b6e69ced316d/src/hotspot/cpu/x86/x86_64.ad#L2750

Reply:
Got it. I submitted new testing.